File: RoadSense-Automation/Appium-Python/utils/logger.py
import logging
import os
import sys

def get_logger(name: str = "RoadSenseAppium") -> logging.Logger:
    logger = logging.getLogger(name)
    if not logger.handlers:
        logger.setLevel(logging.DEBUG)
        
        logs_dir = os.path.join(os.path.dirname(os.path.dirname(__file__)), "logs")
        os.makedirs(logs_dir, exist_ok=True)
        
        file_handler = logging.FileHandler(os.path.join(logs_dir, "appium_execution.log"))
        file_handler.setLevel(logging.DEBUG)
        
        console_handler = logging.StreamHandler(sys.stdout)
        console_handler.setLevel(logging.INFO)
        
        formatter = logging.Formatter(
            '[%(asctime)s] [%(levelname)s] [%(name)s] %(message)s',
            datefmt='%Y-%m-%d %H:%M:%S'
        )
        file_handler.setFormatter(formatter)
        console_handler.setFormatter(formatter)
        
        logger.addHandler(file_handler)
        logger.addHandler(console_handler)
        
    return logger

File: RoadSense-Automation/Selenium-NodeJS/utils/logger.js
const log4js = require("log4js");
const path = require("path");

log4js.configure({
  appenders: {
    file: { type: "file", filename: path.join(__dirname, "../logs/selenium_execution.log") },
    console: { type: "console" }
  },
  categories: {
    default: { appenders: ["file", "console"], level: "info" }
  }
});

const logger = log4js.getLogger("RoadSenseSelenium");
module.exports = logger;
